e2e/specs/mark-patient-deceased.spec.ts

```diff
@@ -0,0 +1,43 @@
+import { expect } from '@playwright/test';
+import { generateRandomPatient, deletePatient, type Patient } from '../commands';
+import { test } from '../core';
+import { MarkPatientDeceasedPage } from '../pages';
+
+let patient: Patient;
+
+test.beforeEach(async ({ api }) => {
+  patient = await generateRandomPatient(api);
+});
+
+test('Mark a patient as deceased', async ({ page }) => {
+  const markPatientDeceasedPage = new MarkPatientDeceasedPage(page);
+  const todayDate = new Date().toLocaleDateString('en-GB').replace(/\//g, '-');
+  const causeOfDeath = 'Neoplasm/cancer';
+
+  await test.step('When I go to the Patient’s chart page', async () => {
+    await markPatientDeceasedPage.goToPatientChart(patient.uuid);
+  });
+
+  await test.step('And I select "Mark patient deceased" from the Actions menu', async () => {
+    await markPatientDeceasedPage.openMarkDeceasedForm();
+  });
+
+  await test.step('Then I should see a form to enter death details', async () => {
+    await expect(markPatientDeceasedPage.deathDetailsForm()).toBeVisible();
+    await expect(markPatientDeceasedPage.dateOfDeathInput()).toBeVisible();
+    await expect(markPatientDeceasedPage.causeOfDeathRadio(causeOfDeath)).toBeVisible();
+  });
+
+  await test.step('When I fill out the death details and save', async () => {
+    await markPatientDeceasedPage.fillDeathDetails(todayDate, causeOfDeath);
+    await markPatientDeceasedPage.saveAndClose();
+  });
```

Comment:
Use the exact test steps from the ticket. Generic steps can be confusing since they don't show what information was filled in the form.